src/v/archival/ntp_archiver_service.cc

```diff
@@ -541,4 +542,68 @@ uint64_t ntp_archiver::estimate_backlog_size() {
     return total_size;
 }
 
+ss::future<std::optional<cloud_storage::partition_manifest>>
+ntp_archiver::maybe_truncate_manifest(retry_chain_node& rtc) {
+    ss::gate::holder gh(_gate);
+    retry_chain_logger ctxlog(archival_log, rtc, _ntp.path());
+    vlog(ctxlog.info, "archival metadata cleanup started");
+    std::optional<cloud_storage::partition_manifest> result;
```

Comment:
nit: this may be moved after the for loop, it shadows the inner result on line 559.